src/plugins/controls/public/control_types/range_slider/range_slider.component.tsx

```diff
@@ -45,16 +45,19 @@ export const RangeSliderComponent: FC<Props> = ({ componentStateSubject }) => {
     componentStateSubject.getValue()
   );
 
-  const { value = ['', ''], id, title } = useEmbeddableSelector((state) => state);
+  const { value, id, title } = useEmbeddableSelector((state) => state);
 
   const [selectedValue, setSelectedValue] = useState<RangeValue>(value || ['', '']);
 
+  useEffect(() => {
```

Comment:
In order to trigger a re-render when the value changes via `resetSelections`, I had to split the previous `useCallback` in to two steps rather than one - the `onChangeComplete` callback now simply dispatches the new value via `selectRange`, and the `useEffect` listens for changes to `value` in order to force render.

Without this, only manual changes to the selections would trigger a render - `resetSelections` would be ignored because it had no way of communicating to the component.

Reply:
Using a `UseEffect` and a `useState` here seems a bit overcomplicated for the use case. Maybe we should double check with @cqliu1 when she's back, but I was able to get the same behaviour by removing the `useState` and `useEffect` here entirely and just using the value directly from the embeddable input instead of mirroring it:

```
<RangeSliderPopover
      ... other stuff

      value={value ?? ['', '']}
/>
```

Because the slider itself can't handle null / undefined, notice how I've initialized the default value.

Reply:
Ahhh, that's totally fair. I implemented these changes for both the time and range sliders in 3839fbaa77ec2a436a2e5f505f9211121e71dee9 - as long as @cqliu1 and @crob611 don't see any unintended consequences with this, we'll go with it! 👍

Reply:
Awesome, thanks for implementing this. And yeah it'll be good to check with them because I could've potentially not covered an edge case with my testing of it. It _seems_ fine, but they might know better!